services/venice-server/src/main/java/com/linkedin/venice/listener/ReadQuotaEnforcementHandler.java

```diff
@@ -328,7 +330,10 @@ private void updateQuota(PartitionAssignment partitionAssignment) {
           clock);
 
       if (rateLimiter != v) {
-        stats.setNodeQuotaResponsibility(storeName, (long) Math.ceil(quotaInRcu * thisNodeQuotaResponsibility));
+        storeVersionNodeResponsibility.put(topic, (long) Math.ceil(quotaInRcu * thisNodeQuotaResponsibility));
```

Comment:
Can we return early from `updateQuota` for non-current versions?

Reply:
I think we could since we are also calling `onCustomizedViewChange` for every version in `handleStoreChanged`. Do you see any other potential holes where we can result in a current version change without having the appropriate token bucket updated and ready?

Reply:
On a second thought I think it's better to update as often as we can in case of potential race when local store is updated with new current version and with new request coming in but `handleStoreChanged` hasn't finished yet and we cannot find the corresponding new token bucket. The update cost of some in-memory data structure should be negligible especially when partition assignment don't change that frequently.